### PR TITLE
Add new stat rocksdb.table.open.prefetch.tail.read.bytes, rocksdb.table.open.prefetch.tail.{miss|hit}

### PR DESCRIPTION
**Pull request by hx235**
**Context/Summary:**
We are adding new stats to measure behavior of prefetched tail size and look up into this buffer

The stat collection is done in FilePrefetchBuffer but only for prefetched tail buffer during table open for now using FilePrefetchBuffer enum. It's cleaner than the alternative of implementing in upper-level call places of FilePrefetchBuffer for table open. It also has the benefit of extensible to other types of FilePrefetchBuffer if needed. See db bench for perf regression concern.

**Test:**
**- Piggyback on existing test** 
**- rocksdb.table.open.prefetch.tail.miss is harder to UT so I manually set prefetch tail read bytes to be small and run db bench.** 
```
./db_bench -db=/tmp/testdb -statistics=true -benchmarks="fillseq" -key_size=32 -value_size=512 -num=5000 -write_buffer_size=655 -target_file_size_base=655 -disable_auto_compactions=false -compression_type=none -bloom_bits=3  -use_direct_reads=true
```
```
rocksdb.table.open.prefetch.tail.read.bytes P50 : 4096.000000 P95 : 4096.000000 P99 : 4096.000000 P100 : 4096.000000 COUNT : 225 SUM : 921600
rocksdb.table.open.prefetch.tail.miss COUNT : 91
rocksdb.table.open.prefetch.tail.hit COUNT : 1034
```
**- No perf regression observed in db_bench**

SETUP command: create same db with ~900 files for pre-change/post-change.
```
./db_bench -db=/tmp/testdb -benchmarks="fillseq" -key_size=32 -value_size=512 -num=500000 -write_buffer_size=655360  -disable_auto_compactions=true -target_file_size_base=16777216 -compression_type=none
```
TEST command 60 runs or til convergence: as suggested by @anand1976 and @akankshamahajan15, vary `seek_nexts` and `async_io` in testing.
```
./db_bench -use_existing_db=true -db=/tmp/testdb -statistics=false -cache_size=0 -cache_index_and_filter_blocks=false -benchmarks=seekrandom[-X60] -num=50000 -seek_nexts={10, 500, 1000} -async_io={0|1} -use_direct_reads=true
```
async io = 0, direct io read = true

  | seek_nexts = 10, 30 runs | seek_nexts = 500, 12 runs | seek_nexts = 1000, 6 runs
-- | -- | -- | --
pre-post change | 4776 (± 28) ops/sec;   24.8 (± 0.1) MB/sec | 288 (± 1) ops/sec;   74.8 (± 0.4) MB/sec | 145 (± 4) ops/sec;   75.6 (± 2.2) MB/sec
post-change | 4790 (± 32) ops/sec;   24.9 (± 0.2) MB/sec | 288 (± 3) ops/sec;   74.7 (± 0.8) MB/sec | 143 (± 3) ops/sec;   74.5 (± 1.6) MB/sec

async io = 1, direct io read = true
  | seek_nexts = 10, 54 runs | seek_nexts = 500, 6 runs | seek_nexts = 1000, 4 runs
-- | -- | -- | --
pre-post change | 3350 (± 36) ops/sec;   17.4 (± 0.2) MB/sec | 264 (± 0) ops/sec;   68.7 (± 0.2) MB/sec | 138 (± 1) ops/sec;   71.8 (± 1.0) MB/sec
post-change | 3358 (± 27) ops/sec;   17.4 (± 0.1) MB/sec  | 263 (± 2) ops/sec;   68.3 (± 0.8) MB/sec | 139 (± 1) ops/sec;   72.6 (± 0.6) MB/sec

### CODE REVIEW

**Review by ajkr — DISMISSED** [review]
LGTM, thanks!

**Comment by hx235** [comment]
@ajkr - not official review request but just want to let you know that after adding `rocksdb.table.open.prefetch.tail.{miss|hit}`, I realized there is a cleaner solution to rocksdb.table.open.prefetch.tail.read.bytes and a bug I made in the original PR. I decided to fix them in the same PR but will request again later since the original PR is accepted.

**Comment by hx235** [comment]
@ajkr - Hi it's ready for review! Sorry I figured the benefit of making changes to previous approved PR including the new stats as they are related. For that I have to "dismissed" your previous approval  and request a new review on the PR. Let me know if there is any concern/questions.

**Comment by akankshamahajan15** [comment]
FilePrefetchBuffer side changes LGTM.

**Comment by hx235** [comment]
rebase and re-import (cuz I forgot to do that previously :P)

File: file/file_prefetch_buffer.cc
Comment (ajkr) [inline file/file_prefetch_buffer.cc:649]: This is instrumenting in too many places. Can you do something like revert changes in this function, rename this function TryReadFromCacheUntracked, and use it as follows:  

```
... TryReadFromCache(...) {
  bool ret = TryReadFromCacheUntracked(...);
  if (usage_ == FilePrefetchBufferUsage::kTableOpenPrefetchTail) {
    if (ret) {
      RecordTick(stats_, TABLE_OPEN_PREFETCH_TAIL_HIT);
    } else {
      RecordTick(stats_, TABLE_OPEN_PREFETCH_TAIL_MISS);
    }
  }
  return ret;
}
```

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

File: file/file_prefetch_buffer.cc
Comment (hx235) [reply file/file_prefetch_buffer.cc:649]: Good idea 💡

Comment (hx235) [reply file/file_prefetch_buffer.cc]: Fixed.

File: file/file_prefetch_buffer.h
Comment (hx235) [reply file/file_prefetch_buffer.h]: Fixed

**Pull request closed without merging**
